gateway/DEBUG_GUIDE.md

```
@@ -225,3 +235,287 @@ Send a request to the deployed API:
 curl http://localhost:8080/petstore/v1/pets
 ```
+
+---
+
+## Option 2B: Local Process Debug — Controller + Policy Engine + Python Executor in VS Code
+
+This extends **Option 2A** by also running the Python Executor on the host, giving you full debugger access to the Python policy runtime.
+
+> [!NOTE]
+> **When to use this:** You are developing or debugging a Python policy and need to set breakpoints, add print statements, or iterate rapidly without rebuilding Docker images.
+
```

markdownlint-cli2 (0.22.1)

[warning] Blank line inside blockquote

(MD028, no-blanks-blockquote)